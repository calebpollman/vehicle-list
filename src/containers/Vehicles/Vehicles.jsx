import React, {Component} from 'react';
import {connect} from 'react-redux';
import {vehiclesGetData} from '../../actions/vehicles';

class Vehicles extends Component {
  constructor(props) {
    super(props);
    
    this.state = {}
  }

  componentDidMount() {
    this.props.vehiclesGetData();
  }

  render() {
    console.log(this.props);
    return (
      <div>Vehicles</div>
    );
  }
};

const mapStateToProps = state => {
  return {
    vehicles: state.vehicles,
    vehiclesHasError: state.vehiclesHasError,
    vehiclesIsLoading: state.vehiclesIsLoading,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    vehiclesGetData: () => dispatch(vehiclesGetData())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Vehicles);