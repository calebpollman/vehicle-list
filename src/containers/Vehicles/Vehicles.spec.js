import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import Vehicles from './Vehicles';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const initialState = {};
const mockStore = configureStore(middlewares);

describe('Vehicles', () => {
  let wrapper, store;
  
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Vehicles store={store} />).dive();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render a wrapper <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should contain the <SearchBar />', () => {
    expect(wrapper.find('SearchBar').length).toEqual(1);
  });
});

describe('search', () => {
  let wrapper, store;
  
  const vehicles = [
    {
      "year" : "2013",
      "make" : "Kia",
      "model" : "Optima",
    },
    {
      "year" : "2013",
      "make" : "Hyundai",
      "model" : "Accent",
    },
    {
      "year" : "2016",
      "make" : "Lincoln",
      "model" : "MKX",
    },
  ];

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Vehicles store={store} />).dive();
    wrapper.setState({vehicles});
  });

  it('returns an array of objects', () => {
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(Array.isArray(searchResults)).toBe(true);
    expect(searchResults[0]).toBeTruthy();
    expect(typeof searchResults[0]).toBe('object');
  });

  it('returns an array of vehicles based on make', () => {
    wrapper.setState({searchTerm: 'Kia'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([{
      "year" : "2013",
      "make" : "Kia",
      "model" : "Optima",
    }]);
  });

  it('returns an array of vehicles based on model', () => {
    wrapper.setState({searchTerm: 'Optima'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([{
      "year" : "2013",
      "make" : "Kia",
      "model" : "Optima",
    }]);
  });

  it('returns an array of vehicles based on year', () => {
    wrapper.setState({searchTerm: '2013'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([
      {
        "year" : "2013",
        "make" : "Kia",
        "model" : "Optima",
      },
      {
        "year" : "2013",
        "make" : "Hyundai",
        "model" : "Accent",
      },
    ]);
  });

  it('search terms can be in any order', () => {
    wrapper.setState({searchTerm: 'Hyundai 2013 Accent'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([
      {
        "year" : "2013",
        "make" : "Hyundai",
        "model" : "Accent",
      },
    ]);
  });

  it('will return vehicles with a partial value', () => {
    wrapper.setState({searchTerm: 'Hyu'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([
      {
        "year" : "2013",
        "make" : "Hyundai",
        "model" : "Accent",
      },
    ]);
  });

  it('returns an empty array if no results', () => {
    wrapper.setState({searchTerm: 'Hyuz'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults.length).toEqual(0);
  });

  it('works with uppercase search terms', () => {
    wrapper.setState({searchTerm: 'HYUNDAI'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([
      {
        "year" : "2013",
        "make" : "Hyundai",
        "model" : "Accent",
      },
    ]);
  });

  it('works with lowercase search terms', () => {
    wrapper.setState({searchTerm: 'hyundai'});
    wrapper.instance().updateVehicleList();
    const searchResults = wrapper.state('searchResults');
    expect(searchResults).toEqual([
      {
        "year" : "2013",
        "make" : "Hyundai",
        "model" : "Accent",
      },
    ]);
  });
});
