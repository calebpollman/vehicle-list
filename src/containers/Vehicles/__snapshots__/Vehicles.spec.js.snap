// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
    }
    vehicles={undefined}
    vehiclesGetData={[Function]}
    vehiclesHasError={undefined}
    vehiclesIsLoading={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBar
          searchActive={false}
          toggleSearchActive={[Function]}
          updateSearchTerm={[Function]}
/>,
        <SortBar
          sortOptions={
                    Array [
                              "Year",
                              "Make",
                              "Model",
                            ]
          }
          sortSelection="year"
          updateSortSelection={[Function]}
/>,
        <List
          listLoading={true}
          searchActive={false}
          searchResults={Array []}
          vehicles={Array []}
/>,
      ],
      "className": "view-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchActive": false,
          "toggleSearchActive": [Function],
          "updateSearchTerm": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "sortOptions": Array [
            "Year",
            "Make",
            "Model",
          ],
          "sortSelection": "year",
          "updateSortSelection": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "listLoading": true,
          "searchActive": false,
          "searchResults": Array [],
          "vehicles": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBar
            searchActive={false}
            toggleSearchActive={[Function]}
            updateSearchTerm={[Function]}
/>,
          <SortBar
            sortOptions={
                        Array [
                                    "Year",
                                    "Make",
                                    "Model",
                                  ]
            }
            sortSelection="year"
            updateSortSelection={[Function]}
/>,
          <List
            listLoading={true}
            searchActive={false}
            searchResults={Array []}
            vehicles={Array []}
/>,
        ],
        "className": "view-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchActive": false,
            "toggleSearchActive": [Function],
            "updateSearchTerm": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "sortOptions": Array [
              "Year",
              "Make",
              "Model",
            ],
            "sortSelection": "year",
            "updateSortSelection": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "listLoading": true,
            "searchActive": false,
            "searchResults": Array [],
            "vehicles": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
